### PR TITLE
added object identification functionality into language service

### PR DESCRIPTION
**Pull request by laurennathan**
Added code in Language Service to Handle Identification Requests.
Given a sql object in a script, it parses the script and returns the full object identifier as a string

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        /// <summary>
+        /// Position of the Identification request
+        /// </summary>
+        public Position Position { get; set;  }          
+
+        /// <summary>
+        /// Position of the Identification request
+        /// </summary>
+        public string word { get; set;  }
```
Comment (aasimkhan30) [inline src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs:31]: Nit: we should follow camel case for property names

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        /// </summary>
+        public string DocumentUri { get; set; }     
+
+        /// <summary>
+        /// Position of the Identification request
+        /// </summary>
+        public Position Position { get; set;  }          
+
+        /// <summary>
+        /// Position of the Identification request
```
Comment (aasimkhan30) [inline src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs:29]: Please add the right doc comments

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+using Microsoft.SqlTools.Hosting.Protocol.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices.Contracts
+{
+    /// <summary>
+    /// Parameters to pass into IdentificationRequest
+    /// </summary>
+    public class IdentificationParams
```
Comment (aasimkhan30) [inline src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs:16]: Maybe DetectSqlObjectTypeParams? @kburtram what do you think? Identification seems to vague to indicate what this api is actually doing

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        /// <summary>
+        /// Position of the Identification request
+        /// </summary>
+        public string word { get; set;  }         
+    }
+
+    public class IdentificationRequest
+    {
+        public static readonly
+            // Takes in a position and document uri, returns the object identification as a list of strings: [server, database, schema, objectname]
```
Comment (aasimkhan30) [inline src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs:37]: It's just returning a string and not a list of strings.

File: src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            RequestType<IdentificationParams, string> Type =
```
Comment (aasimkhan30) [inline src/Microsoft.SqlTools.ServiceLayer/LanguageServices/Contracts/Identification.cs:38]: Shouldn't you return something more descriptive? 
Something that includes
- ObjectId
- Name
- ObjectType

**Comment by Charles-Gagnon** [comment]
Bulk closing stale PRs, please update and reopen if you plan on continuing this work!

**Pull request closed without merging**